homeassistant/components/moon/sensor.py

```diff
@@ -13,7 +13,16 @@
 
 DEFAULT_NAME = "Moon"
 
-ICON = "mdi:brightness-3"
+MOON_ICONS = {
+    STATE_NEW_MOON: "mdi:moon-new",
+    STATE_WAXING_CRESCENT: "mdi:moon-waxing-crescent",
+    STATE_FIRST_QUARTER: "mdi:moon-first-quarter",
+    STATE_WAXING_GIBBOUS: "mdi:moon-waxing-gibbous",
+    STATE_FULL_MOON: "mdi:moon-full",
+    STATE_WANING_GIBBOUS: "moon-waning-gibbous",
+    STATE_LAST_QUARTER: "mdi:moon-last-quarter",
+    STATE_WANING_CRESCENT: "mdi:moon-waning-crescent"
+}
```

Comment:
The state constants are not defined.

Reply:
since this is my first pr, I based myself on season.sensor, and missed that , sorry.

would this be what is needed?

```
STATE_NEW_MOON: "new_moon",
STATE_WAXING_CRESCENT: "waxing_crescent",
STATE_FIRST_QUARTER: "first_quarter",
STATE_WAXING_GIBBOUS: "waxing_gibbous",
STATE_FULL_MOON: "full_moon",
STATE_WANING_GIBBOUS: "waning_gibbous",
STATE_LAST_QUARTER: "last_quarter",
STATE_WANING_CRESCENT
```

hope you will help me get to understand all this...

Reply:
added state constants

Reply:
I think this won't work as if I'm not mistaken self._state contains a number (i.e. the number of days after a new moon mod 28), not the state names.

Reply:
yes, thats what I thought hence my own expressed doubt about the correct call. need to get to the state(self) but don't know how to formulate the final call for that.
please help me?

Reply:
I think you could return MOON_ICONS[state] but I'm no expert at all, maybe @fabaff could suggest!

Reply:
`self.state` returns a string which could be used for the lookup.

Now looking at the code, especially at the naming of the states, I would say that my initial idea was to delegate the selection of the icon to the frontend.

Reply:
thanks for getting back.
to finalize this PR then, what would I need to use for the final line:

```
return MOON_ICONS.get(self.state, "mdi:brightness-3")
```
?
not really sure what you mean by delegating this to the frontend though, seems to me making this native in the backend is preferable to any kind of templating I the frontend (which is what I do know...)
```    sensor.moon:
      templates:
        icon: >
          if (state === 'full_moon') return 'mdi:moon-full';
          if (state === 'new_moon') return 'mdi:moon-new';
          return 'mdi:moon-' + state.replace('_','-');
```